rclpy/rclpy/executors.py

```diff
@@ -521,7 +510,8 @@ def _wait_for_ready_callbacks(
                 entity_count += waitable.get_num_entities()
 
             # Construct a wait set
-            with _WaitSet() as wait_set, ExitStack() as context_stack:
+            wait_set = _rclpy.rclpy_get_zero_initialized_wait_set()
+            with ExitStack() as context_stack:
```

Comment:
Now that the GC can destroy the wait set when it pleases, are there any dependencies between the wait set and the entities that we need to handle?

Reply:
None that needs to be handled by the wait set. `context_stack` makes sure the entities in the wait set stay alive until `rcl_wait()` finishes. After that it's totally fine to fini the wait set even if entities no longer exist because it doesn't do anything with those pointers.